qa/vagrant/src/main/java/org/elasticsearch/packaging/util/Packages.java

```diff
@@ -0,0 +1,266 @@
+ */
+
+package org.elasticsearch.packaging.util;
+
+import org.elasticsearch.packaging.util.Shell.Result;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import static org.elasticsearch.packaging.util.FileMatcher.Fileness.Directory;
+import static org.elasticsearch.packaging.util.FileMatcher.Fileness.File;
+import static org.elasticsearch.packaging.util.FileMatcher.file;
+import static org.elasticsearch.packaging.util.FileMatcher.p644;
+import static org.elasticsearch.packaging.util.FileMatcher.p660;
+import static org.elasticsearch.packaging.util.FileMatcher.p750;
+import static org.elasticsearch.packaging.util.FileMatcher.p755;
+import static org.elasticsearch.packaging.util.FileUtils.getCurrentVersion;
+import static org.elasticsearch.packaging.util.FileUtils.getDistributionFile;
+import static org.elasticsearch.packaging.util.Platforms.isSysVInit;
+import static org.elasticsearch.packaging.util.Platforms.isSystemd;
+import static org.elasticsearch.packaging.util.ServerUtils.waitForElasticsearch;
+import static org.hamcrest.CoreMatchers.anyOf;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class Packages {
+
+    public static final Path SYSVINIT_SCRIPT = Paths.get("/etc/init.d/elasticsearch");
+    public static final Path SYSTEMD_SERVICE = Paths.get("/usr/lib/systemd/system/elasticsearch.service");
+
+    public static void assertInstalled(Distribution distribution) {
+        final Result status = packageStatus(distribution);
+        assertThat(status.exitCode, is(0));
+
+        Platforms.onDPKG(() -> assertFalse(Pattern.compile("(?m)^Status:.+deinstall ok").matcher(status.stdout).find()));
+    }
+
+    public static void assertRemoved(Distribution distribution) {
+        final Result status = packageStatus(distribution);
+
+        Platforms.onRPM(() -> assertThat(status.exitCode, is(1)));
+
+        Platforms.onDPKG(() -> {
+            assertThat(status.exitCode, anyOf(is(0), is(1)));
+            if (status.exitCode == 0) {
+                assertTrue(Pattern.compile("(?m)^Status:.+deinstall ok").matcher(status.stdout).find());
+            }
+        });
+    }
+
+    public static Result packageStatus(Distribution distribution) {
+        final Shell sh = new Shell();
+        final String manager;
+        final Result result;
+
+        if (distribution.packaging == Distribution.Packaging.RPM) {
+            manager = "rpm";
+            result = sh.runIgnoreExitCode("rpm -qe " + distribution.flavor.name);
+        } else {
+            manager = "dpkg";
+            result = sh.runIgnoreExitCode("dpkg -s " + distribution.flavor.name);
+        }
+
+        if (result.exitCode > 1) {
```

Comment:
Why is it ok to ignore exit code  1 but not the others ?

Reply:
This ended up like this after some refactoring and doesn't really make sense now, so I'll remove it. Originally it was that I didn't want to conflate exit code 1 (package doesn't exist) with some other error from the package manager